### PR TITLE
Move to Spring MVC for APIs

### PR DESCRIPTION
**Pull request by tgianos**

### CODE REVIEW

**Comment by ajoymajumdar** [comment]
Good job on taking the effort to make the changes. Lets review it on Monday in our meeting.

File: metacat-common/src/main/java/com/netflix/metacat/common/exception/MetacatException.java [hunk 1]
```diff
@@ -21,91 +21,39 @@
...
- * Created by amajumdar on 3/27/15.
+ * Base exception for Metacat errors exposed externally.
+ *
+ * @author amajumdar
+ * @author tgianos
  */
-public class MetacatException extends WebApplicationException {
+public class MetacatException extends RuntimeException {
     private static final MetacatJson METACAT_JSON = MetacatJsonLocator.INSTANCE;
     private static final ObjectNode EMPTY_ERROR = METACAT_JSON.emptyObjectNode().put("error", "");
```
Comment (danielbwatson) [inline metacat-common/src/main/java/com/netflix/metacat/common/exception/MetacatException.java]: Now unused.

File: metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * Handle Genie Exceptions.
+     *
+     * @param response The HTTP response
+     * @param e        The exception to handle
+     * @throws IOException on error in sending error
+     */
+    @ExceptionHandler(MetacatException.class)
+    public void handleGenieException(
```
Comment (danielbwatson) [inline metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java]: handleMetacatException also the comments reference Genie

File: metacat-client/src/main/java/com/netflix/metacat/client/module/MetacatErrorDecoder.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public Exception decode(final String methodKey, final Response response) {
                 message = Util.toString(response.body().asReader());
                 try {
                     final ObjectNode body = METACAT_JSON.parseJsonObject(message);
-                    message = body.path("error").asText();
+                    message = body.path("message").asText();
```
Comment (danielbwatson) [inline metacat-client/src/main/java/com/netflix/metacat/client/module/MetacatErrorDecoder.java]: This is my only actual concern.  Unless you can get everyone to update to the latest client at the same time you release this version of Metacat, the existing clients will work with the new server they just won't see the error messages.

**Comment by tgianos** [comment]
@ajoymajumdar @zhljen I believe I've addressed all the feedback. Can one/both of you review again and approve so I can merge this monstrosity?

**Pull request merged**